Assets/Scripts/Algorithms/Planning/LaunchAngleInterpolator.cs

```diff
@@ -33,15 +33,69 @@ public LaunchAngleOutput Plan(in LaunchAngleInput input) {
                                  timeToPosition: interpolatedDataPoint.Data[1]);
   }
 
-  // Get the intercept position.
+  // Get the intercept position for a target from the default origin (0,0,0).
+  //
+  // INTERCEPT CALCULATION PRINCIPLE:
+  // The intercept position is where both the interceptor and target should meet.
+  // For a successful intercept, this should be very close to the predicted target position.
+  // The interpolation table provides launch parameters (angle, time) to reach that position,
+  // not a different intercept location.
+  //
+  // This is the original implementation preserved for backward compatibility.
+  //   position: Target position where intercept should occur
+  // Returns: Calculated intercept position (should be very close to input position)
   public Vector3 GetInterceptPosition(Vector3 position) {
+    if (_interpolator == null) {
+      InitInterpolator();
+    }
+
     Vector2 direction = ILaunchAnglePlanner.ConvertToDirection(position);
     Interpolator2DDataPoint interpolatedDataPoint =
         _interpolator.Interpolate(direction[0], direction[1]);
     Vector3 cylindricalPosition = Coordinates3.ConvertCartesianToCylindrical(position);
-    return Coordinates3.ConvertCylindricalToCartesian(r: interpolatedDataPoint.Coordinates[0],
-                                                      azimuth: cylindricalPosition.y,
-                                                      height: interpolatedDataPoint.Coordinates[1]);
+
+    // For realistic intercepts, use original height. For extreme cases where interpolation
+    // gives very different coordinates, preserve the interpolated distance behavior
```

Comment:
I am hesitant to make this change because it seems to be a responsibility of the interpolator subclass instead of having the parent perform this check.  E.g., nearest neighbor interpolation with large granularity will naturally have more error than a smooth 2D interpolation.